aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -44,9 +84,26 @@ struct aesd_buffer_entry *aesd_circular_buffer_find_entry_offset_for_fpos(struct
 */
 void aesd_circular_buffer_add_entry(struct aesd_circular_buffer *buffer, const struct aesd_buffer_entry *add_entry)
 {
-    /**
-    * TODO: implement per description 
-    */
+    struct aesd_buffer_entry* aesd_buff_entry = (struct aesd_buffer_entry*) malloc(sizeof(struct aesd_buffer_entry));
+    
+    memcpy(aesd_buff_entry, add_entry, sizeof(*add_entry));
+
+    if(buffer == NULL)
+    {
+        return;
+    }
+        
+    buffer->entry[buffer->in_offs].buffptr = aesd_buff_entry->buffptr;
+    buffer->entry[buffer->in_offs].size = aesd_buff_entry->size;
```

Comment:
@rajatchaple where are you freeing `aesd_buff_entry` ?  Also what purpose does this have?  Why not just set directly with `add_entry` passed?